superset/connectors/sqla/utils.py

```diff
@@ -137,7 +139,7 @@ def get_virtual_table_metadata(dataset: "SqlaTable") -> List[ResultSetColumnType
     try:
         with closing(engine.raw_connection()) as conn:
             cursor = conn.cursor()
-            query = dataset.database.apply_limit_to_sql(statements[0])
+            query = dataset.database.apply_limit_to_sql(statements[0], limit=1)
```

Comment:
bycatch: fetch virtual table should limit query.

Reply:
Nice improvement 👍